test/repl.js

```diff
@@ -44,9 +44,16 @@ exports["Repl"] = {
       debug: false
     });
 
+    // Extra-careful guard against calling test.done() twice here.
+    // This was causing "Cannot read property 'setUp' of undefined" errors
+    // See https://github.com/caolan/nodeunit/issues/234
+    var calledTestDone = false;
     var reallyExit = this.sandbox.stub(process, "reallyExit", function() {
       reallyExit.restore();
-      test.done();
+      if (!calledTestDone) {
+        calledTestDone = true;
+        test.done();
+      }
```

Comment:
This was a little harder to track down, but my first clue was [this issue](https://github.com/caolan/nodeunit/issues/198) on the nodeunit repo, followed by [this issue](https://github.com/caolan/nodeunit/issues/234) that is still open and summarizes the problem well; nodeunit raises this really opaque error if you happen to call `test.done()` twice in a single test.

I scoured rgb.collection.js for places where this might happen but eventually noticed that it always failed _right_ at the top of that file... So I checked the test file before it, and sure enough, there's some tricky and system-dependent callback stuff going on here.  I did some more extra logging on a separate branch and discovered that the `"reallyExit"` stub was getting called twice in failure cases (examples [here](https://travis-ci.org/code-dot-org/johnny-five/jobs/224146923#L1135) and [here](https://travis-ci.org/code-dot-org/johnny-five/jobs/224141898#L1134)).  I'm not sure how - that doesn't seem possible, since we restore the stub when it's called - but I figured a second guard against calling `test.done()` twice might be in order.

If anyone has a better idea of how the `reallyExit` stub is getting called a second time, I'm all ears.  Maybe something in the previous test?